accepted/targeting-packs/targeting-packs.md

```diff
@@ -0,0 +1,149 @@
+  (ie on non-Windows OS's). On Windows, we will probably make the .NET SDK
+  continue to rely on the reference assemblies from the targeting packs by
+  default.
+* For the package IDs, I suggest the following:
+  - For the metapackage: `Microsoft.NETFramework.ReferenceAssemblies`
+  - For the version-specific packages:
+    `Microsoft.NETFramework.ReferencesAssemblies.net462` (where net462 is
+    replaced with the corresponding NuGet short framework Identifier)
+* The NuGet packages will include a `.targets` file that sets the
+  `TargetFrameworkRootPath` to a path within the NuGet package. This will allow
+  the existing MSBuild logic in the `GetReferenceAssemblyPaths` target and other
+  logic such as automatically referencing the facades if necessary to continue
+  to work as normal.
+* The logic in the `.targets` file in the NuGet packages will only set the
+  `TargetFrameworkRootPath` if the `TargetFrameworkIdentifier` and
+  `TargetFrameworkVersion` properties match the reference assemblies that the
+  package provides. It will also only set the property if the
+  `UseReferenceAssembliesFromPackage` property is set to true.
+* The reference assembly packages should include the `.xml` intellisense
+  documentation files. I believe the `.xml` documentation files have to be in
+  the same folder as the reference assemblies, so there isn't a way to deliver
+  them as a separate package.
+* If we want to support localized intellisense, we would need to create a
+  separate set of packages and corresponding meta-package for each language
+  supported. The IDs of these packages could follow the pattern
+  `Microsoft.NETFramework.ReferenceAssemblies.pt-br`. The SDK could use a
+  property to select which language's metapackage to reference.
+* The reference assembly packages should not show up as dependencies of "normal"
+  packages. Thus, the reference assembly packages should set
+  `developmentDependency` to true in it's metadata. Likewise, when the .NET SDK
+  automatically references the reference assembly metapackage, it should use
+  `PrivateAssets="All"`.
+* The reference assembly packages should include the same layout of files that
+  are installed by the targeting packs under `C:\Program Files (x86)\Reference
+  Assemblies\Microsoft\Framework`. This should be rooted at the path specified
+  by the `TargetFrameworkRootPath` property in the package. For example, if the
+  `.targets` file, which is in the `build` folder of the NuGet package, sets the
+  `TargetFrameworkRootPath` to `$(MSBuildThisFileDirectory)`, then the .NET
+  4.6.2 reference assembly package should have the reference assemblies and
+  intellisense files in the `build\.NETFramework\v4.6.2` folder, as well as have
+  the `Facades`, `PermissionSets`, and `RedistList` folders with corresponding
+  files under that folder.
+* The version number of each package should start at 1.0.0. When a new version
```

Comment:
Was any thought given to just release the net462 reference assemblies in a `Microsoft.NETFramework.ReferenceAssemblies, v4.6.2` package, and to have a separate `Microsoft.NETFramework.ReferenceAssemblies.All` metapackage?

Reply:
I think this would result in (overall) fewer names, easier to find/reason about, etc

Reply:
There have been (rare) occasions where reference assemblies had a bug that was serviced. I think this would be blocked by TFM version == package version.

Reply:
Yes, using the target framework version as the package version would prevent us from servicing the package if we needed to.

Also, we wouldn't want someone to manually reference the `Microsoft.NETFramework.ReferenceAssemblies` package, and end up with the wrong version of the package for the version of the framework they're targeting (which could happen if they retarget their project, or update to a newer package version via the NuGet UI, for example).

Reply:
Yes, each package feels like it's a separate "product" to me. The package targeting .NET 4.5.1 can't be used if you're trying to target .NET 4.5, for example - it's not just "a patched version of the same thing". It's a package you'd use for a different purpose.

So +1 to "start at 1.0.0".

Reply:
The same could be said for .NETStandard Library and any future servicing fixes that come out for it...

I would assume that whatever plan is required there would also apply here.

I don't think that these are separate products at all, they are simply the reference assemblies for different versions of the .NET Framework

Reply:
> Netstandard.Library package version is not 1:1 with .NETStandard TFM version either.

Yep, and the fact that we tried to "look" close to the TFM version has caused more problem than it solved. `1.0.0` is the way to go.